### ISSUE TITLE
Hybrid-AR generate() never honours the stop token — conditioning carries post-stop frames (ComfyUI's async stop-check pattern solves the stated perf concern)

### ISSUE THREAD

**Issue by aidiffuser** [ThuGie/Music-3-Lora#7]
## What happens

`trainer/music3/hybrid_ar.py` `generate()` computes the stop signal but never acts on it (~line 397):

```python
is_stop = code_or_stop == 0
# Keep generating to decode_limit. Training cond must match audio length;
# checking stop with .item() also synced the GPU every frame (~8 min/45s chunk).
c0 = torch.where(is_stop, torch.zeros_like(code_or_stop), code_or_stop - 1).repeat(2)
```

The only `break` in the loop is the length cap. ComfyUI's `comfy/ldm/minimax_music/ar.py` breaks on the stop token. We diffed the two `generate()` bodies line by line — everything else is essentially identical; termination is the only divergence.

## Consequence

Every conditioning sequence (training chunks, uncond, and sample prompts) carries off-distribution frames generated past the point the AR considered the song finished. For a 45s window whose "planned song" ends early, the tail of the conditioning is post-stop garbage the base model never produces in normal inference. This degrades both training conditioning quality and sample fidelity, and plausibly contributes a hard ceiling on sample quality that more training steps cannot fix.

## Why the fix is not just adding `break` — and how ComfyUI solved the same problem

The comment's two rationales are real: (a) conditioning length must match audio length for training, and (b) per-frame `.item()` checks sync the GPU (~8 min/45s chunk).

ComfyUI solves (b) with an **async pinned-copy pattern**: copy the sampled code to pinned CPU memory with a CUDA event, check the *previous* frame's code one iteration later (`pending_code`/`pending_event`, ar.py ~L284-295) — stop detection with no per-frame sync. That pattern drops in here.

For (a): after stop fires, instead of continuing generation, pad/repeat a terminal frame (or mask the post-stop region in the cond cache and let the trainer treat it as padding) so length alignment is preserved without feeding post-stop rollouts as if they were real plan.

## Environment

Repo @ 59a728a; verified against ComfyUI master's `minimax_music/ar.py` on the same machine.

---
*Found during the same investigation as #6 (see its correction comment for the full picture); AI-assisted per the README's policy.*
